packages/network-controller/tests/provider-api-tests/helpers.ts

```diff
@@ -430,9 +425,8 @@ export async function waitForPromiseToBeFulfilledAfterRunningAllTimers(
 
   // `hasPromiseBeenFulfilled` is modified asynchronously.
   /* eslint-disable-next-line no-unmodified-loop-condition */
-  while (!hasPromiseBeenFulfilled && numTimesClockHasBeenAdvanced < 15) {
-    clock.runAll();
-    await new Promise((resolve) => originalSetTimeout(resolve, 10));
```

Comment:
Making this change was necessary to get the new provider API tests I'm adding for the failover logic to work without running into timeout issues.